react-frontend-app/src/App.js

```diff
@@ -20,25 +20,61 @@ import BarGraph from './components/BarGraph'
 function App() {
   const [surname, setSurname] = useState('');
   const [zipcode, setZip] = useState('');
-  var prediction = {};
+
+  // ZRP inputs
+  const [firstName, setFirst] = useState('');
+  const [midName, setMid] = useState('');
+  const [precinct, setPrecinct] = useState('');
+  const [gender, setGender] = useState('');
+  const [county, setCounty] = useState('');
+  const [cong, setCong] = useState('');
+  const [senate, setSenate] = useState('');
+  const [house, setHouse] = useState('');
+  const [birth, setBirth] = useState('');
 
   // Functions that are called whenever the text input is changed
   function handleSurname(event) {
     setSurname(event.target.value);
   }
-
   function handleZip(event) {
     setZip(event.target.value);
   }
+  function handlefirstName(event) {
```

Comment:
I think we talked about this, but it would be good to refactor this into a handleSubmit — but the most important thing is to work on getting the data to save properly and consistently.

Reply:
+1 on this. Would be good to work in a refactor commit in the near future that addresses this, without breaking/changing functionality